plugins/texteditor.koplugin/main.lua

```diff
@@ -0,0 +1,498 @@
+        table.insert(sub_item_table, {
+            text = T("%1. %2", i, filename),
+            callback = function()
+                self:checkEditFile(file_path, true)
+            end,
+            hold_callback = function()
+                -- Show full path and some info, and propose to remove from history
+                local text
+                local attr = lfs.attributes(file_path)
+                if attr then
+                    local filesize = util.getFormattedSize(attr.size)
+                    local lastmod = os.date("%Y-%m-%d %H:%M", attr.modification)
+                    text = T(_("File path:\n%1\n\nFile size: %2 bytes\nLast modified: %3\n\nRemove this file from text editor history?"),
+                                file_path, filesize, lastmod)
+                else
+                    text = T(_("File path:\n%1\n\nThis file does not exist anymore.\n\nRemove it from text editor history?"),
+                                file_path)
+                end
+                UIManager:show(ConfirmBox:new{
+                    text = text,
+                    ok_text = _("Yes"),
+                    cancel_text = _("No"),
+                    ok_callback = function()
+                        self:removeFromHistory(file_path)
+                    end,
+                })
+            end,
+        })
+    end
+    return sub_item_table
+end
+
+function TextEditor:removeFromHistory(file_path)
+    for i=#self.history, 1, -1 do
+        if self.history[i] == file_path then
+            table.remove(self.history, i)
+        end
+    end
+    self.last_view_pos[file_path] = nil
+end
+
+function TextEditor:addToHistory(file_path)
+    local new_history = {}
+    table.insert(new_history, file_path)
+    -- Trim history and cleanup duplicates
+    local seen = {}
+    seen[file_path] = true
+    while #self.history > 0 and #new_history < self.history_keep_size do
+        local item = table.remove(self.history, 1)
+        if not seen[item] then
+            table.insert(new_history, item)
+            seen[item] = true
+        end
+    end
+    self.history = new_history
+end
+
+function TextEditor:newFile()
+    self:loadSettings()
+    UIManager:show(ConfirmBox:new{
+        text = _([[To start editing a new file, you will have to:
+
+- First select a directory
+- Then type the new file filename
+- And start editing it
+
+Do you want to proceeed?]]),
+        ok_text = _("Yes"),
+        cancel_text = _("No"),
+        ok_callback = function()
+            local path_chooser = PathChooser:new{
+                select_directory = true,
+                select_file = false,
+                height = Screen:getHeight(),
+                path = self.last_path,
+                onConfirm = function(dir_path)
+                    local file_input
+                    file_input = InputDialog:new{
+                        title =  _("Enter new file filename"),
```

Comment:
new file filename? just new filename

Reply:
Because it's about giving a birth name to a new file. Not about giving a new filename to an existing file (that _new filename_ evokes for me). Will use this alternative so none of us is shocked :)
```
- First select a directory
- Then enter a name for the new file
- And start editing it
title =  _("Enter filename")
```